pkg/model/names.go

```diff
@@ -36,7 +36,7 @@ func (b *KopsModelContext) SecurityGroupName(role kops.InstanceGroupRole) string
 		return "bastion." + b.ClusterName()
 	case kops.InstanceGroupRoleNode:
 		return "nodes." + b.ClusterName()
-	case kops.InstanceGroupRoleMaster:
+	case kops.InstanceGroupRoleMaster, kops.InstanceGroupRoleAPIServer:
 		return "masters." + b.ClusterName()
```

Comment:
This is an interesting move. We could introduce another security group for API servers, and attach it to both control plane machines and API server-only machines, though at the moment I can't think of firewall rules that the former require separately from the latter.

Reply:
The masters security group allows inbound etcd traffic which wouldn't be applicable on apiserver hosts. I believe it is exposed in a wider range than just the etcd ports themselves, so it could potentially impact other use cases. We also allow all traffic between control plane hosts which isn't applicable for dedicated apiserver hosts.

while it would add complexity to this PR (or a followup), given the pain of revoking security group rules down the road I think it would be best to create a dedicated apiserver security group now (or in a followup).

Reply:
This was a short-cut, I must admit.

Right now, I am thinking
1.  merge this
2. create an e2e test that verifies this is working
3. tighten the rules

3 definitely need to happen before beta or we need to featureflag this. I should have plenty of time to fix this before beta though.